### DISCUSSION TITLE
Thoughts on switching the led to red when the battery goes bellow X percent

### DISCUSSION THREAD

**Discussion by Sichroteph** [eduardofilo/RG350_adam_image#81] [Ideas & Requests]
Simplemenu is great for launching a game at boot. However when the emulation app is not retroarch, we loose track on the battery state.

Do you know if we can trigger the led color with a linux command so a simple background script could do the trick ? I know how to retrieve the battery percentage into the filesystem.
It can be another notification like the vibration motor.

If there are no easy way to do this, should I look into Paul Ceucueil code (his shortcut deamons seems quite deep into the system) or directly into the open dingux kernel ? 

I am mainly playing on reGBA, another idea would be to add this info into its system screen.

**Comment by Sichroteph** [comment]
Removed for future research

**Comment by eduardofilo** [comment]
Sorry, but I haven't studied the system so deeply. I compile everything that I am reading and I do not remember something related to the management of the LED. I thought the LED would start flashing when the battery was low. At least stock did. As I hardly play with the console and I have it almost always connected to the computer to access the terminal by SSH, I have not been able to see if ODbeta does the same, but I am almost sure that it does. Doesn't it?

**Comment by Sichroteph** [comment]
Thanks for your answer. I am not sure either, maybe that is a feature that has been lost with open dingux updates. I am doing some tests on a Rogue image - kernel 3.12.75-dingux+ and went to arount 20% without led flashing. Maybe I should go lower to be sure.

**Comment by eduardofilo** [comment]
Well, the same battery charging level (the real physical one) can be interpreted differently in ROGUE than in ODbeta, so the test would have to be taken beyond, yes.

**Reply by Xaumir** [reply]
I've been testing it for the past few days (sorry, this is my first comment on github but I think it may be relevant) and what I've found out on my RG350P is that no battery level indicator works properly on it. I've been playing for the past couple days until RetroArch accused my battery to be at 1%. From then, I've put Alundra 2 to run at maximum brightness and 50% volume, let it alone in a place that used 98% of CPU and it lasted for an hour before the console ran out of juice and died. No blinking led or anything. Also, the test app never shows the battery percentage. It's stuck at 0%, but it does work on ROGUE. I haven't tested ROGUE yet to see if it blinks the led or not, but at least for me, Adam doesn't.

**Reply by eduardofilo** [reply]
Without statistics like those handled by modern smartphones, it is very difficult to have an accurate measurement of the actual capacity remaining in the battery.
The test app was designed for ROGUE and it is not completely adapted to ODbeta, but still useful, and that is why is maintained in Adam.

**Reply by Sichroteph** [reply]
Yes these battery doesn't have a proper Battery Management System that track the exact amount of amperage going out. Voltage based systems are imprecise. On the good side of things, BMS take power on the battery over the months even if the device is off.
